pm-frontend/src/pages/Vault.js

```diff
@@ -61,41 +55,82 @@ export default function PasswordGetter() {
       setLoggedIn(true);
     }
     setAuthToken(localStorage.getItem("authToken"));
+
+    async function getAllSites() {
+      try {
+        const response = await axios.get(
+          "http://localhost:3001/sites",
+          {
+            headers: {
+              token: authToken,
+            },
+          }
+        );
+        setSavedSites(response.data.sites);
+      } catch (error) {
+        console.log(error);
+      }
+    }
+    getAllSites();
+
+    // !testing only
+    // TODO: delete
+    setSavedSites([
+      "Google",
+      "Facebook",
+      "Instagram",
+      "Firefox",
+      "Snapchat",
+      "Reddit",
+      "D2L",
+      "ucalgary",
+      "Miniclip",
+      "Cool Math Games",
+    ])
   }, []);
 
-  const revealPassword = (site) => {
-    console.log("Revealing " + site.label);
-    setRevealed(true);
-    axios.post(
-      "http://localhost:3001/passwords/reveal",
-      {
-        requestedSite: site,
-      },
-      {
-        headers: {
-          token: authToken,
+  const revealPassword = async (site) => {
+    console.log("Revealing " + site);
+    try {
+      const res = await axios.get(
+        "http://localhost:3001/passwords/reveal",
```

Comment:
Endpoint here should be /password?label=${some_label}

Reply:
I changed it, not sure if this is working as intended just based off of vs code syntax highlighting

Reply:
i used back ticks and i think it's good now

Reply:
Yeah back ticks should do it! Sorry forgot to mention